test/OpenTelemetry.Instrumentation.SqlClient.Tests/SqlClientIntegrationTests.cs

```diff
@@ -66,24 +77,37 @@ public void SuccessfulCommandTest(
         var dataSource = sqlConnection.DataSource;
 
         sqlConnection.ChangeDatabase("master");
+        SqlTransaction? transaction = null;
 #pragma warning disable CA2100
         using var sqlCommand = new SqlCommand(commandText, sqlConnection)
 #pragma warning restore CA2100
         {
             CommandType = commandType,
         };
 
+        if (enableTransaction)
+        {
+            transaction = sqlConnection.BeginTransaction();
+            sqlCommand.Transaction = transaction;
+        }
+
+        object commandResult = DBNull.Value;
         try
         {
-            sqlCommand.ExecuteNonQuery();
+            commandResult = sqlCommand.ExecuteScalar();
         }
         catch
         {
         }
 
+        transaction?.Commit();
+
         Assert.Single(activities);
         var activity = activities[0];
 
+#if !NETFRAMEWORK
```

Comment:
Why? Does it mean that this feature will be working only on .NET?
If so, you need to adjust CHANGELOG and the documentation. I think that you should consider also conditional compilation for _production_ code.

Reply:
>Does it mean that this feature will be working only on .NET?

I attempted to implement support for .NET Framework, but it appears there’s no mechanism on the instrumentation side to retrieve the current connection. So for now, we’ll limit support to .NET. 

CHANGELOG and README updated.

> should consider also conditional compilation for production code.

The entire `SqlClientDiagnosticListener.cs` file is wrapped in conditional compilation and won't compile under .NET Framework.